indexing-hadoop/src/main/java/io/druid/indexer/HadoopDruidIndexerMapper.java

```diff
@@ -103,7 +103,9 @@ public final static InputRow parseInputRow(Object value, InputRowParser parser)
   {
     if(parser instanceof StringInputRowParser && value instanceof Text) {
       //Note: This is to ensure backward compatibility with 0.7.0 and before
-      return ((StringInputRowParser)parser).parse(value.toString());
+      return ((StringInputRowParser) parser).parse(value.toString());
+    } else if(value instanceof InputRow) {
+      return (InputRow)value;
```

Comment:
One option is that DatasourceInputFormat returns a "SegmentInputRow" instead of "InputRow" which would only be a wrapper(and extension of InputRow) and we put ugly things like
```
if(row instanceof SegmentInputRow) {
  //serialize using combining aggregators
} else {
  //serialize using normal aggregators
}
```